exporter/datadogexporter/logs_exporter.go

```diff
@@ -114,3 +121,39 @@ func (exp *logsExporter) consumeLogs(ctx context.Context, ld plog.Logs) (err err
 	payloads := exp.translator.MapLogs(ctx, ld)
 	return exp.sender.SubmitLogs(exp.ctx, payloads)
 }
+
+// newLogsAgentExporter creates new instances of the logs agent and the logs agent exporter
+func newLogsAgentExporter(
+	ctx context.Context,
+	params exporter.CreateSettings,
+	cfg *Config,
+	sourceProvider source.Provider,
+) (logsagentpipeline.LogsAgent, exporter.Logs, error) {
+	logComponent := newLogComponent(params.TelemetrySettings)
+	cfgComponent := newConfigComponent(params.TelemetrySettings, cfg)
+	logsAgentConfig := &logsagentexporter.Config{
+		OtelSource:    otelSource,
+		LogSourceName: logSourceName,
+	}
+	hostnameComponent, err := logs.NewHostnameService(ctx, sourceProvider)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initialize logs agent hostname service: %w", err)
+	}
+	logsAgent := logsagentpipelineimpl.NewLogsAgent(logsagentpipelineimpl.Dependencies{
+		Log:      logComponent,
+		Config:   cfgComponent,
+		Hostname: hostnameComponent,
+	})
+	err = logsAgent.Start(ctx)
+	if err != nil {
+		params.Logger.Error("failed to create logs agent", zap.Error(err))
+		return nil, nil, err
+	}
+	pipelineChan := logsAgent.GetPipelineProvider().NextPipelineChan()
+	logsAgentExporter, err := logsagentexporter.NewFactory(pipelineChan).CreateLogsExporter(ctx, params, logsAgentConfig)
+	if err != nil {
+		params.Logger.Error("failed to create logs agent exporter", zap.Error(err))
+		return nil, nil, err
```

Comment:
We don't need to log and return the error, just doing one of the two is enough
```suggestion
		return nil, nil, fmt.Errorf("failed to create logs agent exporter: %w", err)
```